validphys2/src/validphys/commondataparser.py

```diff
@@ -40,21 +108,805 @@
+
+        # Check that plotting.plot_x is being filled
+        if self.plotting.plot_x is None:
+            ermsg = f"No variable selected as x-axis in the plot for {self.name}. Please add plotting::plot_x."
+            if self.plotting.x is not None:
+                ermsg += "If you are using `plotting::x` please change it to `plotting::plot_x`"
+            raise ValidationError(ermsg)
+
+        # Ensure that all variables in the kinematic coverage exist
+        for var in self.kinematic_coverage:
+            if var not in self.kinematics.variables:
+                raise ValidationError(
+                    f"Variable {var} is in `kinematic_coverage` but not included in `kinematics` for {self.name}"
+                )
+
+        if len(self.kinematic_coverage) > 3:
+            raise ValidationError(
+                "Only a maximum of 3 variables can be used for `kinematic_coverage`"
+            )
+
+    def apply_variant(self, variant_name):
+        """Return a new instance of this class with the variant applied
+
+        This class also defines how the variant is applied to the commondata
+        """
+        try:
+            variant = self.variants[variant_name]
+        except KeyError as e:
+            raise ValueError(f"The requested variant does not exist {self.variant_name}") from e
+
+        variant_replacement = {}
+        if variant.data_uncertainties is not None:
+            variant_replacement["data_uncertainties"] = variant.data_uncertainties
+        if variant.theory is not None:
+            variant_replacement["theory"] = variant.theory
+        if variant.data_central is not None:
+            variant_replacement["data_central"] = variant.data_central
+
+        return dataclasses.replace(self, applied_variant=variant_name, **variant_replacement)
+
+    @property
+    def is_positivity(self):
+        return self.setname.startswith("NNPDF_POS")
+
+    @property
+    def is_integrability(self):
+        return self.setname.startswith("NNPDF_INTEG")
+
+    @property
+    def is_lagrange_multiplier(self):
+        return self.is_positivity or self.is_integrability
+
+    @property
+    def path_data_central(self):
+        return self._parent.folder / self.data_central
+
+    def load_data_central(self):
+        """Loads the data for this commondata returns a dataframe
+
+        Returns
+        -------
+        pd.DataFrame
+            a dataframe containing the data
+        """
+        if self.is_lagrange_multiplier:
+            data = np.zeros(self.ndata)
+        else:
+            datayaml = _quick_yaml_load(self.path_data_central)
+            data = datayaml["data_central"]
+        data_df = pd.DataFrame(data, index=range(1, self.ndata + 1), columns=["data"])
+        data_df.index.name = _INDEX_NAME
+        return data_df
+
+    @property
+    def paths_uncertainties(self):
+        return [self._parent.folder / i for i in self.data_uncertainties]
```

Comment:
I maybe missing the rational here but why should all the uncertainty variants be loaded once?

Reply:
This is a list to allow people to separate uncertainties. This was requested at some point in the past and have stayed like that. I don't think anybody used it.

The idea was that you would have a number of uncertainties that are shared between variants, and then some that are unique, then you can have variants like:

```
my_variant_one:
    uncertainties: 
         - file_shared
         - file_unique_one
my_variant_two:
    uncertainties: 
         - file_shared
         - file_unique_two
```

Reply:
I see, it is good to have such flexibility.